src/DataStructures/ModalVector.hpp

```diff
@@ -0,0 +1,88 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include "DataStructures/DataVector.hpp"  // IWYU pragma: keep
+#include "DataStructures/VectorImpl.hpp"
+#include "Utilities/PointerVector.hpp"  // IWYU pragma: keep
+#include "Utilities/TMPL.hpp"
+
+// IWYU pragma: no_include <blaze/math/expressions/DVecMapExpr.h>
+// IWYU pragma: no_include <blaze/math/typetraits/IsVector.h>
+// IWYU pragma: no_include "DataStructures/DataVector.hpp"
+// IWYU pragma: no_forward_declare ConstantExpressions_detail::pow
+
+/*!
+ * \ingroup DataStructuresGroup
+ * \brief A class for storing spectral coefficients on a mesh.
+ *
+ * \details A ModalVector holds an array of spectral coefficients, and can be
+ * either owning (the array is deleted when the ModalVector goes out of scope)
+ * or non-owning, meaning it just has a pointer to an array.
+ *
+ * Only basic mathematical operations are supported with ModalVectors. In
+ * addition to addition, subtraction, multiplication, division, there
```

Comment:
This sentence makes it sound like multiplying two `ModalVector`s is allowed, which I don't think it is (or should be). A different way to do the dox would be to write:

```cpp
The following mathematical binary expressions are allowed:
- ModalVector-ModalVector subtraction
- ModalVector-ModalVector addition
- double-ModalVector multiplication
- double-ModalVector division
- ModalVector-ModalVector max
- ModalVector-ModalVector min
```

I'm not sure how you feel about that but it might be a bit more explicit.